pkg/credentials/filter/secret_detection.go

```diff
@@ -0,0 +1,89 @@
+package filter
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// DetectedSecret represents a secret that was detected in the entrypoint.
+type DetectedSecret struct {
+	Name  string
+	Value []byte
```

Comment:
The docs say that the values of the secrets are not transmitted, but it looks like an example DetectedSecret would be Name: "password", Value: "my-super-secret-password"-- am I understanding this correctly? Is the idea here that the entrypoint is reading the secret files to get the contents of the secrets and prevent them from being printed?

Reply:
There are currently to processes in place which could make this a little bit confusing at first. But its quite simple:

- Secret extraction runs on controller side and extracts the environment variables names and file paths where secrets are store in a pod
- Secret detection runs on the entrypoint side and uses the extracted secret locations to actually detect the secret values from the environment variables and files.

This also means that a "detected" secret on entrypoint side contains the full secret that needs to be redacted by the entypoint. But this secret value is not transmitted to it, just the location. The secret value itself comes either from an environment variable or file content.